File: src/api/feeds.ts
import { type Fetcher, defaultFetcher } from "@literate.ink/utilities";
import { Request } from "~/core/request";
import { decodeFeed } from "~/decoders/feed";
import type { Feed } from "~/models";

export const feeds = async (fetcher: Fetcher = defaultFetcher): Promise<Array<Feed>> => {
  const request = new Request("feeds.json");
  const { content } = await fetcher(request);
  const { results } = JSON.parse(content);

  return results.map(decodeFeed);
};

File: library/src/commonMain/kotlin/ink/literate/crowous/api/feeds.kt
package ink.literate.crowous.api

import ink.literate.crowous.core.Request
import ink.literate.crowous.decoders.decodeFeed
import kotlinx.serialization.json.*
import ink.literate.crowous.models.Feed

suspend fun feeds (): List<Feed> {
    val request = Request("feeds.json")
    val response = request.send()
    val content = Json.parseToJsonElement(response)
    val results = content.jsonObject["results"]!!.jsonArray

    return results.map { json -> decodeFeed(json.jsonObject) }
}
